### ISSUE TITLE
linter: react/rules-of-hooks should report invalid useEffectEvent escapes

### ISSUE THREAD

**Issue by KentoMoriwaki** [oxc-project/oxc#23744]
### What version of Oxlint are you using?

1.70.0

### What command did you run?

oxlint -c .oxlintrc.json

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["react"],
  "rules": {
    "react/rules-of-hooks": "error"
  }
}
```

### What happened?

`react/rules-of-hooks` does not appear to report some invalid `useEffectEvent` usages that are reported by `eslint-plugin-react-hooks`.

Minimal reproduction:

```tsx
import { useEffectEvent } from "react";

function MyComponent({ theme }) {
  const onClick = useEffectEvent(() => {
    showNotification(theme);
  });

  return <Child onClick={onClick} />;
}
```

Expected: this should be reported because an Effect Event is passed outside the component/hook where it was created.

Another related case:

```tsx
import { useEffectEvent } from "react";

function useCustomHook() {
  const onEvent = useEffectEvent(() => {});

  return { onEvent };
}
```

Expected: this should be reported because the Effect Event escapes the custom hook.

React docs say that Effect Events can only be called from inside Effects or other Effect Events, and should not be passed to other components or hooks:

[https://react.dev/reference/react/useEffectEvent](https://react.dev/reference/react/useEffectEvent)

`eslint-plugin-react-hooks` has test coverage for these cases in `ESLintRulesOfHooks-test.js`:

[https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/__tests__/ESLintRulesOfHooks-test.js](https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/__tests__/ESLintRulesOfHooks-test.js)

Related upstream React fix for inline JSX prop escapes:

[https://github.com/facebook/react/issues/34793](https://github.com/facebook/react/issues/34793)
[https://github.com/facebook/react/pull/34820](https://github.com/facebook/react/pull/34820)

`oxlint` already has support for `useEffectEvent` in `react/exhaustive-deps` via:

[https://github.com/oxc-project/oxc/pull/14041](https://github.com/oxc-project/oxc/pull/14041)

This issue is specifically about `react/rules-of-hooks` enforcing the `useEffectEvent` escape/call-site restrictions.
